birdhouse/pavics-compose.sh

```diff
@@ -146,6 +147,11 @@ if [ x"$1" = x"up" ]; then
   docker volume create jupyterhub_data_persistence  # jupyterhub db and cookie secret
   docker volume create thredds_persistence  # logs, cache
 
+  if [ ! -d "$GEOSERVER_DATA_DIR" ]; then
```

Comment:
So this directory cannot be created on the host before launching geoserver, else the permissions will not be good?
And given that we give the ownership to 1000:10001 from the container, we could prevent access directly from the host... Not saying it's a bad thing, maybe just for backup purposes it could be more complicated.

Reply:
The root cause is the new geoserver runs as user with id 1000:10001 so the data folder needs to have the same ownership, else geoserver can not write to it.  And given the geoserver container do not run as root, it can not switch the ownership itself.

Now, when docker comes up, if the dir do not already exist, it will create it with ownership 0:0.

So this code basically ensure the dir creation is not done by docker but before hand with proper ownership.

Hopefully do not affect backup since backup processes probably run as root to have read-only on everything (the other stuff under `/data/` mostly owned by root).  And I hope backup is a read-only and not write so should not accidentally create a new file with wrong ownership.